src/clojure/nrepl/server.clj

```diff
@@ -64,6 +64,31 @@
                           s))
                       % %))))
 
+(defn no-auth-wrapper
+  "Allows all requests and warns to *err* unless
+  NREPL_SUPPRESS_DEFAULT_AUTH_WARNING is set to true."
+  [handle-message]
+  (when-not (some-> (System/getenv "NREPL_SUPPRESS_DEFAULT_AUTH_WARNING")
+                    (= "true"))
+    (binding [*out* *err*]
+      (println "nREPL set to require no authorization; allowing all access")
+      (println "Suppress warning with NREPL_SUPPRESS_DEFAULT_AUTH_WARNING=true")))
+  handle-message)
+
+(defn file-auth-wrapper
```

Comment:
That's named `file-auth-wrapper`, but I don't see it doing anything with files.

Reply:
Oh, right, that's vestigial -- I'll fix it in the next revision.